tests/scripts/deploy-validate-vault.sh

```diff
@@ -206,8 +206,20 @@ function set_up_vault_kubernetes_auth {
   # create the RBAC for this SA
   kubectl -n "$ROOK_NAMESPACE" create clusterrolebinding vault-tokenreview-binding --clusterrole=system:auth-delegator --serviceaccount="$ROOK_NAMESPACE":"$ROOK_VAULT_SA"
 
+  kubectl apply -f - <<EOF
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: rook-vault-auth-secret
+  annotations:
+    kubernetes.io/service-account.name: vault-auth
+type: kubernetes.io/service-account-token
+EOF
+
   # get the service account common.yaml created earlier
-  VAULT_SA_SECRET_NAME=$(kubectl -n "$ROOK_NAMESPACE" get sa "$ROOK_VAULT_SA" -o jsonpath="{.secrets[*]['name']}")
+  VAULT_SA_SECRET_NAME=$(kubectl get secrets --output=json |
+    jq -r '.items[].metadata | select(.name|startswith("vault-auth-")).name')
```

Comment:
these are the only changes for vault(since this file has other changes coming from linter so pointing it out)